test/model-definition.test.js

```diff
@@ -257,6 +257,35 @@ describe('ModelDefinition class', function() {
     done();
   });
 
+  it('should report errors for deep query than maxDepthOfQuery', function(done) {
+    var MyModel = memory.createModel('my-model', {}, {
+      maxDepthOfQuery: 5,
+    });
+
+    var filter = {where: {and: [{and: [{and: [{and: [{and: [{and:
```

Comment:
nitpick: In the future, if we have more tests like these, we can extract this part out to a helper function like `givenFilterWithDepth()` or something like that for better readability.